ceno_emul/src/platform.rs

```diff
@@ -9,33 +9,42 @@ use crate::addr::{Addr, RegIdx};
 /// - codes of environment calls.
 #[derive(Clone, Debug)]
 pub struct Platform {
-    pub rom: Range<Addr>,
-    pub ram: Range<Addr>,
+    pub prog_code: Range<Addr>,
+    pub prog_data: Option<HashSet<Addr>>,
+    pub stack: Range<Addr>,
+    pub heap: Range<Addr>,
     pub public_io: Range<Addr>,
     pub hints: Range<Addr>,
-    pub stack_top: Addr,
     /// If true, ecall instructions are no-op instead of trap. Testing only.
     pub unsafe_ecall_nop: bool,
 }
 
 pub const CENO_PLATFORM: Platform = Platform {
-    rom: 0x2000_0000..0x3000_0000,
-    ram: 0x8000_0000..0xFFFF_0000,
+    prog_code: 0x2000_0000..0x3000_0000,
+    prog_data: None, // This is an `Option` to allow `const` here.
```

Comment:
Move the comment to the type definition for Platform, please.  (Or change to `BTreeSet` and avoid the problem.)